addons/account/wizard/account_automatic_entry_wizard.py

```diff
@@ -0,0 +1,387 @@
+    _name = 'account.automatic.entry.wizard'
+    _description = 'Create Automatic Entries'
+
+    # General
+    action = fields.Selection([('accrual', 'Change Period'), ('transfer', 'Change Account')], required=True)
+    data = fields.Text(compute="_compute_data")
+    preview_data = fields.Text(compute="_compute_preview_data")
+    move_line_ids = fields.Many2many('account.move.line')
+    date = fields.Date(required=True, default=fields.Date.today())
+    company_id = fields.Many2one('res.company', required=True, readonly=True)
+    company_currency_id = fields.Many2one('res.currency', related='company_id.currency_id')
+    percentage = fields.Float("Percentage", compute='_compute_percentage', readonly=False, store=True)
+    total_amount = fields.Monetary(compute='_compute_total_amount', store=True, readonly=False, currency_field='company_currency_id')
+    journal_id = fields.Many2one('account.journal', required=True, readonly=False, string="Journal",
+        domain="[('company_id', '=', company_id), ('type', '=', 'general')]",
+        related="company_id.automatic_entry_default_journal",
+        help="Journal where to create the entry.")
+
+    # Accrual
+    account_type = fields.Selection([('income', 'Revenue'), ('expense', 'Expense')], compute='_compute_total_amount', required=True)
+    expense_accrual_account = fields.Many2one('account.account', readonly=False,
+        domain="[('company_id', '=', company_id), ('internal_type', 'not in', ('receivable', 'payable'))]",
+        related="company_id.expense_accrual_account_id")
+    revenue_accrual_account = fields.Many2one('account.account', readonly=False,
+        domain="[('company_id', '=', company_id), ('internal_type', 'not in', ('receivable', 'payable'))]",
+        related="company_id.revenue_accrual_account_id")
+
+    # Transfer
+    destination_account_id = fields.Many2one(string="To", comodel_name='account.account', help="Account to transfer to.")
+    display_currency_helper = fields.Boolean(string="Currency Conversion Helper", compute='_compute_display_currency_helper', help="Technical field used to indicate whether or not to display the tooltip informing a currency conversion will be performed with the transfer")
+
+    @api.constrains('percentage', 'action')
+    def _constraint_percentage(self):
+        for record in self:
+            if not (0.0 < record.percentage <= 100.0):
+                raise UserError(_("Percentage must be between 0 and 100"))
+            if record.percentage != 100 and record.action != 'accrual':
+                raise UserError(_("Percentage can only be set for Change Period method"))
+
+    @api.depends('percentage', 'move_line_ids')
+    def _compute_total_amount(self):
+        for record in self:
+            percentage = record.percentage or 100
+            total_amount = percentage * sum(record.move_line_ids.mapped(lambda l: (l.debit + l.credit))) / 100
+            if record.total_amount != total_amount:
```

Comment:
Because otherwise there is a loop messing up the recompute, and we don't get the expected results.
```python
    @api.depends('percentage', 'move_line_ids')
    def _compute_total_amount(self):
        ...
```
```python
    @api.depends('total_amount', 'move_line_ids')
    def _compute_percentage(self):
        ...
```